cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -179,9 +182,31 @@ func main() {
 		provisionerOptions = append(provisionerOptions, controller.AdditionalProvisionerNames([]string{supportsMigrationFromInTreePluginName}))
 	}
 
+	var csiNodeLister storagelisters.CSINodeLister
+	var factory informers.SharedInformerFactory
+	if utilfeature.DefaultFeatureGate.Enabled(features.Topology) {
+		// Create informer to prevent hit the API server for all resource request
+		factory = informers.NewSharedInformerFactory(clientset, ctrl.ResyncPeriodOfCsiNodeInformer)
```

Comment:
if the lister is only necessary when topology feature is used, should it use `supportsTopology()` instead to capture all conditions?

https://github.com/kubernetes-csi/external-provisioner/blob/b9613492f7c958d9ba615f71dcbe64d392a9888b/pkg/controller/controller.go#L657

Reply:
Do you mean to change the condition `utilfeature.DefaultFeatureGate.Enabled(features.Topology)` with `pluginCapabilities[csi.PluginCapability_Service_VOLUME_ACCESSIBILITY_CONSTRAINTS] &&
		utilfeature.DefaultFeatureGate.Enabled(features.Topology)` or do you mean to use `csiNodeLister` as initiated inside the `csiProvisioner` with the `supportsTopology()` check instead of a external provided property of the struct?

Reply:
I added additional check for plugin capabilities.

Reply:
Yup this is what I meant, though in its current form, I'm worried that someone later on will only change one place but not the other. It would be good to have a shared utility function instead in a different file (maybe package/util)

Reply:
Makes sense, thanks. Putting the function inside `topology.go` instead of util package makes sense to me.